### ISSUE TITLE
scim-schema should be included as a dependency, not copied

### ISSUE THREAD

**Issue by tkent** [osiam/osiam#197]
(This is half an issue, half a question)

It's unclear why the scim-schema project's code is copied into this one and not included as a dependency.

Unless it's the case that the scim-schema project is being retired (which doesn't appear to be true), then it seems like the osiam server is an ideal place to showcase it's use.

If I'm missing something, please do point me in the right direction!

**Comment by tkent** [comment]
(nevermind, noticed the depreciated message at the top of the project.)

**Comment by wallner** [comment]
Hello @tkent,

maybe some explanation is in order: We started out with the `scim-schema` as a common view on scim resources as java objects for both the java client and the OSIAM servers. It turned out that while this was not a bad idea to get OSIAM off the ground, it is not that smart in the long run for several reasons, for instance that server and client do have slightly different ideas about how scim resources should be represented in java or the immutable schema is a very good thing to have in the connector (the client) but makes it use clunky and cumbersome in the server. Also the only consumer of the `scim-schema` known to  us is actually our client. For that reason we decided to copy the scim-schema into the connector and the server for the upcoming 3.0 release.

I hope I could shed some light.
Florian
